grow/pods/rendered_document.py

```diff
@@ -0,0 +1,39 @@
+"""Tracks the rendered document information."""
+
+import hashlib
+import os
+
+
+class RenderedDocument(object):
+    """Keeps track of the information for the rendered document."""
+
+    def __init__(self, path, content, tmp_dir=None):
+        if isinstance(content, unicode):
+            content = content.encode('utf-8')
+
+        self.path = path
+        self.hash = hashlib.sha1(content).hexdigest()
+        self.tmp_dir = tmp_dir
+
+        if tmp_dir:
+            with open(self.filename, "w") as tmp_file:
+                tmp_file.write(content)
+        else:
+            self._content = content
+
+    @property
+    def content(self):
```

Comment:
I wonder if we should prefer more of a file-based API so that we can eventually optimize further with buffered reads and writes. We don't necessarily need to do the latter now, but initially we could potentially support that in the future by having a few methods like:

- open -> returns a file descriptor (if no tmp_dir is provided, returns a cStringIO instance)
- write -> opens the file descriptor and writes to it
- read -> opens the file descriptor and reads the entire contents

this might change the initialization a bit, so instead of passing contents directly to rendereddoc, you'd do:

```
doc = RenderedDoc(path, tmp_dir=foo)
doc.write(content)
```

Then i think to the caller, it'll feel more clear that when they call `doc.read()` that it's being read from the filesystem, whereas accessing attributes like `doc.content` always feels like it's stored in memory to me.

Reply:
I've updated it to use a `write` and `read` functions. For now I'm going to forgo the `open` until we actually need it.